src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LanguageServerProjectLoader.cs

```diff
@@ -146,100 +152,125 @@ private async ValueTask LoadOrReloadProjectsAsync(ImmutableSegmentedList<Project
         }
     }
 
+    protected abstract Task<(RemoteProjectFile? projectFile, bool hasAllInformation, BuildHostProcessKind preferred, BuildHostProcessKind actual)> TryLoadProjectAsync(
```

Comment:
doc?

it is called tryLoad, but what's the failure mode here?

Reply:
i think i would prefer this return a nullable tuple vs a tuple, with a nullable first entry?